lib/clouds/shared_functions.py

```diff
@@ -813,13 +822,342 @@ def populate_interface(self, obj_attr_list) :
+    @trace
+    def is_cloud_image_uuid(self, imageid) :
+        '''
+        TBD
+        '''
+        if imageid == "to_replace" :
+            return False
+        
+        if self.get_description() == "Amazon Elastic Compute Cloud" :
+            if len(imageid) > 4 :
+                if imageid[0:4] == "ami-" :
+                    if is_number(imageid[5:], True) :
+                        return True
+
+        if self.get_description() == "Cloudbench SimCloud" or self.get_description() == "Cloudbench NoOpCloud" :
+            if len(imageid) == 36 and imageid.count('-') == 4 :
+                return True
+        
+        if self.get_description() == "OpenStack Cloud" :
+            if len(imageid) == 36 and imageid.count('-') == 4 :
+                return True
+
+        if self.get_description() == "SoftLayer Cloud" :
+            if len(imageid) == 7 and is_number(imageid) :
+                return True
+
+        if self.get_description() == "Google Compute Engine" :
+            if len(imageid) == 19 and is_number(imageid) :
+                return True
+
+        if self.get_description() == "DigitalOcean Cloud" :
+            if len(imageid) == 8 and is_number(imageid) :
+                return True
+
+        if self.get_description() == "Parallel Docker Manager Cloud" :
+            if len(imageid) == 64 and is_number(imageid, True) :
+                return True
+
+        if self.get_description() == "Kubernetes Cloud" :
+            return True
+            if len(imageid) == 64 and is_number(imageid, True) :
+                return True
+
+        if self.get_description() == "Parallel Container Manager Cloud" :
+            if len(imageid) == 64 and is_number(imageid, True) :
+                return True
+        
+        return False
+
+    @trace
+    def check_ssh_key(self, vmc_name, key_name, vm_defaults, internal = False, connection = None) :
+        '''
+        TBD
+        '''
+
+        _key_pair_found = False      
+        
+        if not key_name :
+            _key_pair_found = True
+        else :
+            _msg = "Checking if the ssh key pair \"" + key_name + "\" is created"
+            _msg += " on VMC " + vmc_name + "...."
+            if not internal :
+                cbdebug(_msg, True)
+            else :
+                cbdebug(_msg)            
+                
+            _pub_key_fn = vm_defaults["credentials_dir"] + '/'
+            _pub_key_fn += vm_defaults["ssh_key_name"] + ".pub"
+
+            _key_type, _key_contents, _key_fingerprint = get_ssh_key(_pub_key_fn, self.get_description())
+
+            vm_defaults["pubkey_contents"] = _key_type + ' ' + _key_contents
+
+            if not _key_contents :
+                _fmsg = _key_type 
+                cberr(_fmsg, True)
+                return False
+            
+            _key_pair_found = False
+
+            _registered_key_pairs = {}
+            if self.get_description() == "Cloudbench SimCloud" or \
```

Comment:
Same thing here. We can't have each individual cloud's name hard-coded in here like this. These need to into the individual adapters so that the maintainer of each adapter only needs to be responsible for what's in their head.